finetune/target_encoders.py

```diff
@@ -68,7 +68,58 @@ def transform(self, y):
         labels = super().transform(y)
         return self._make_one_hot(labels)
 
+class OrdinalRegressionEncoder(OrdinalEncoder, BaseEncoder):
+    
+    def __init__(self):
+        self.num_outputs = None
+        super().__init__()
+        
+    def fit(self, x):
+        x = np.array(x)
+        rank = len(x.shape)
+        if rank == 1:
+            x = np.expand_dims(x, 1)
+        self.fit_transform(x)
+        return self
+
+    def transform(self, x):
+        x = np.array(x)
+        rank = len(x.shape)
+        if rank == 1:
+            x = np.expand_dims(x, 1)  # for single output value regression.
+        labels = super().transform(x)
+        labels = self.rank_to_thresholds(labels)
+        return labels
+        
+    def fit_transform(self, x):
+        super().fit(x)
+        labels = self.transform(x)
+        self.num_outputs = labels.shape[1]
```

Comment:
`fit_transform` should call `fit` and `transform`, rather than `fit` calling `fit_transform`